ms2deepscore/data_generators.py

```diff
@@ -10,150 +11,188 @@
+            equal frequencies. Default is set to [(0, 0.5), (0.5, 1)], which means
+            that pairs with scores <=0.5 will be picked as often as pairs with scores
+            > 0.5.
+        augment_peak_removal_max
+            Maximum fraction of peaks (if intensity < below augment_peak_removal_intensity)
+            to be removed randomly. Default is set to 0.2, which means that between
+            0 and 20% of all peaks with intensities < augment_peak_removal_intensity
+            will be removed.
+        augment_peak_removal_intensity
+            Specifying that only peaks with intensities < max_intensity will be removed.
         augment_intensity
             Change peak intensities by a random number between 0 and augment_intensity.
             Default=0.1, which means that intensities are multiplied by 1+- a random
             number within [0, 0.1].
 
         """
-        self.spectrums_binned_dicts = spectrums_binned_dicts
-        self.score_array = y_true
-        self.score_array[np.isnan(y_true)] = 0
-        self.training_ids = training_ids
+        assert score_array.shape[0] == score_array.shape[1] == len(inchikey_score_mapping), \
+            f"Expected score_array of size {len(inchikey_score_mapping)}x{len(inchikey_score_mapping)}."
+        assert len(inchikeys_all) == len(spectrums_binned), \
+            "inchikeys_all and spectrums_binned must have the same dimension."
+        self.spectrums_binned = spectrums_binned
+        self.score_array = score_array
+        #self.score_array[np.isnan(score_array)] = 0
+        self.inchikey_ids = self.__exclude_nans(inchikey_ids)
+        self.inchikey_score_mapping = inchikey_score_mapping
+        self.inchikeys_all = inchikeys_all
         self.dim = dim
         self.batch_size = batch_size
-        self.num_turns = num_turns #number of go's through all IDs
+        self.num_turns = num_turns
         self.peak_scaling = peak_scaling
         self.shuffle = shuffle
         self.ignore_equal_pairs = ignore_equal_pairs
         self.on_epoch_end()
-        self.inchikey_mapping = inchikey_mapping
-        self.inchikeys_array = inchikeys_array
         self.same_prob_bins = same_prob_bins
-        self.augment_peak_removal = augment_peak_removal
+        self.augment_peak_removal_max = augment_peak_removal_max
+        self.augment_peak_removal_intensity = augment_peak_removal_intensity
         self.augment_intensity = augment_intensity
 
     def __len__(self):
         """Denotes the number of batches per epoch"""
         # TODO: this means we don't see all data every epoch, because the last half-empty batch
-        #  is omitted. I guess that is expected behavior?
-        return int(self.num_turns) * int(np.floor(len(self.training_ids) / self.batch_size))
+        #  is omitted. I guess that is expected behavior? --> Yes, with the shuffling in each epoch that seem OK to me (and makes the code easier).
```

Comment:
Cool :) I guess you can remove the comments now? Or maybe add a small note to the docstring.

Reply:
Indeed. Will remove that comment.